### PR TITLE
sci-libs/torchvision: add 0.17.1

### PR DESCRIPTION
**Pull request by littlewu2508**
Include a patch for ffmpeg 6.0 fix

Limit setuptools thread to 1, while ninja respect MAKEOPTS

### CODE REVIEW

File: sci-libs/torchvision/torchvision-0.17.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+RDEPEND="
+	$(python_gen_cond_dep '
+		dev-python/numpy[${PYTHON_USEDEP}]
+		dev-python/typing-extensions[${PYTHON_USEDEP}]
+		dev-python/pillow[${PYTHON_USEDEP}]
+		dev-python/requests[${PYTHON_USEDEP}]
+		dev-python/scipy[${PYTHON_USEDEP}]
+	')
+	sci-libs/pytorch[${PYTHON_SINGLE_USEDEP}]
+	media-video/ffmpeg
```
Comment (AndrewAmmerlaan) [inline sci-libs/torchvision/torchvision-0.17.1.ebuild:30]: Do we need a `:=` slot operator here for `ffmpeg`?

File: sci-libs/torchvision/torchvision-0.17.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	dev-qt/qtcore:5
```
Comment (AndrewAmmerlaan) [inline sci-libs/torchvision/torchvision-0.17.1.ebuild:31]: Is it also compatible with qt6?

File: sci-libs/torchvision/torchvision-0.17.1.ebuild (hunk 1 above)
Comment (littlewu2508) [reply sci-libs/torchvision/torchvision-0.17.1.ebuild:30]: Indeed. I checked the linking and one of the library links to all three: libavutil libavcodec libavformat.

File: sci-libs/torchvision/torchvision-0.17.1.ebuild (hunk 2 above)
Comment (littlewu2508) [reply sci-libs/torchvision/torchvision-0.17.1.ebuild:31]: I unmerged dev-qt/qtcore and torchvision built and ran without issue. So I suppose qt is not a necessary dependency. Not sure why it is introduced in previous version.

@AndrewAmmerlaan In this repo I saw all previous torchvision commits are authorized by you. Do you have any clue on why qt is introduced?

Comment (AndrewAmmerlaan) [reply sci-libs/torchvision/torchvision-0.17.1.ebuild:31]: I honestly can't remember, looking through the history it seems it was always there. But perhaps it is not needed. Is the size of the install tree the same when you compile with qtcore on the system versus without?

Comment (littlewu2508) [reply sci-libs/torchvision/torchvision-0.17.1.ebuild:31]: >  Is the size of the install tree the same when you compile with qtcore on the system versus without

Yes, exactly the same.

Comment (AndrewAmmerlaan) [reply sci-libs/torchvision/torchvision-0.17.1.ebuild:31]: Then it is probably unnecessary indeed

**Comment by AndrewAmmerlaan** [comment]
I'm getting some error in the test phase:
```
ImportError while loading conftest '/tmp/portage/sci-libs/torchvision-0.17.1/work/vision-0.17.1/test/conftest.py'.
test/conftest.py:7: in <module>
from common_utils import (
test/common_utils.py:22: in <module>
from torchvision import io, tv_tensors
torchvision/__init__.py:6: in <module>
from torchvision import _meta_registrations, datasets, io, models, ops, transforms, utils
torchvision/_meta_registrations.py:163: in <module>
@torch._custom_ops.impl_abstract("torchvision::nms")
/usr/lib/python3.11/site-packages/torch/library.py:440: in inner
handle = entry.abstract_impl.register(func_to_register, source)
/usr/lib/python3.11/site-packages/torch/_library/abstract_impl.py:30: in register
if torch._C._dispatch_has_kernel_for_dispatch_key(self.qualname, "Meta"):
E   RuntimeError: operator torchvision::nms does not exist
```

This looks otherwise okay to me.

**Comment by littlewu2508** [comment]
> I'm getting some error in the test phase:
> 
> ```
> ImportError while loading conftest '/tmp/portage/sci-libs/torchvision-0.17.1/work/vision-0.17.1/test/conftest.py'.
> test/conftest.py:7: in <module>
> from common_utils import (
> test/common_utils.py:22: in <module>
> from torchvision import io, tv_tensors
> torchvision/__init__.py:6: in <module>
> from torchvision import _meta_registrations, datasets, io, models, ops, transforms, utils
> torchvision/_meta_registrations.py:163: in <module>
> @torch._custom_ops.impl_abstract("torchvision::nms")
> /usr/lib/python3.11/site-packages/torch/library.py:440: in inner
> handle = entry.abstract_impl.register(func_to_register, source)
> /usr/lib/python3.11/site-packages/torch/_library/abstract_impl.py:30: in register
> if torch._C._dispatch_has_kernel_for_dispatch_key(self.qualname, "Meta"):
> E   RuntimeError: operator torchvision::nms does not exist
> ```
> 
> This looks otherwise okay to me.

I think it's related to the testing environment.

After emerging torchvision into system, copy `${S}/test/`  to `/tmp`, and install https://pypi.org/project/expecttest/ (I quickly packaged it), I can ran tests (but it will take very long).

**Comment by littlewu2508** [comment]
Let's disable the tests currently like `sc-libs/pytorch`, and then work on testing of pytorch and torchvision together. This includes making `dev-python/expecttest` to ::gentoo and executing the test properly without running into import issues.

**Pull request closed without merging**